lib/reporting/reports/orders_and_fulfillment/order_cycle_customer_totals.rb

```diff
@@ -163,6 +167,23 @@ def distributor_customer(line_items)
           user = line_items.first.order.user
           user&.customer_of(distributor)
         end
+
+        def voucher_label(order)
+          return '' unless voucher_applicable?(order)
+
+          voucher = order.voucher_adjustments.take.originator
+          voucher&.code.to_s # in case if we don't get the voucher, return ""
+        end
+
+        def voucher_amount(order)
+          return '' unless voucher_applicable?(order)
```

Comment:
I am wondering if actually need this check ? 

If I am not mistaken `(order.total - order.pre_discount_total).abs` would return 0 if no voucher has been applied to the order.

Reply:
Yes, but for consistency between voucher label vs amount's blank state, I added this check to return empty strings as their blank state value. Thanks. :)